File: code/search/linear_search/LinearSearch.scala
/* Part of Cosmos by OpenGenus Foundation */

object LinearSearch extends App {

  def linearSearch[A: Equiv](list: List[A], element: A): Option[Int] = {
    list match {
      case Nil => None
      case head :: tail => if (head == element) Some(0) else (linearSearch(tail, element).map(_ + 1))
    }
  }

  val list = (0 to 1000) map (_ => scala.util.Random.nextInt(1000))
  val element = scala.util.Random.nextInt(1000)
  val output = linearSearch(list.toList, element)
  Console.println(s"Index: $output")
  
}

File: JAVA prog/LinearSearch.java
import java.util.*;
public class LinearSearch {
    private static final Scanner sc = new Scanner(System.in);
    public static void main(String[]args) {
        int n = sc.nextInt();
        int arr[] = new int[n];
        for(int i=0; i<n; i++)
            arr[i] = sc.nextInt();
        int ele = sc.nextInt();
        for(int i=0; i<n; i++) {
            if(ele == arr[i])
                System.out.println("Element found at "+ (i+1) + " position");
        }
    }
}
